scripts/actions/spells/white/sacrifice.lua

```diff
@@ -10,25 +11,48 @@ end
 spellObject.onSpellCast = function(caster, target, spell)
     local removables = { xi.effect.FLASH, xi.effect.BLINDNESS, xi.effect.PARALYSIS, xi.effect.POISON, xi.effect.CURSE_I, xi.effect.CURSE_II, xi.effect.DISEASE, xi.effect.PLAGUE }
 
-    -- remove one effect and add it to me
-    for i, effect in ipairs(removables) do
-        if target:hasStatusEffect(effect) then
-            spell:setMsg(xi.msg.basic.MAGIC_ABSORB_AILMENT)
+    -- Maximum number of effects to transfer
+    local maxTransferCount = caster:hasStatusEffect(xi.effect.AFFLATUS_SOLACE) and 7 or 1
+
+    -- Use the helper function to transfer status effects
+    local transferCount = 0  -- Initialize transferCount
 
-            local statusEffect = target:getStatusEffect(effect)
+    -- Loop over the original effects
+    for _, effect in ipairs(removables) do
+        if transferCount < maxTransferCount then
+            if target:hasStatusEffect(effect) then
+                local statusEffect = target:getStatusEffect(effect)
 
-            -- only add it to me if I don't have it
-            if not caster:hasStatusEffect(effect) then
-                caster:addStatusEffect(effect, statusEffect:getPower(), statusEffect:getTickCount(), statusEffect:getDuration())
+                -- only add it to me if I don't have it
+                if not caster:hasStatusEffect(effect) then
+                    caster:addStatusEffect(effect, statusEffect:getPower(), statusEffect:getTickCount(), statusEffect:getDuration())
+                    transferCount = transferCount + 1
+                end
+
+                target:delStatusEffect(effect)
             end
+        else
+            break  -- Break the loop if the maximum number of effects are transferred
+        end
+    end
 
-            target:delStatusEffect(effect)
-            return 1
+    -- Check if the maximum transfer count is not reached
+    while transferCount < maxTransferCount do
+        local erasableEffect = caster:stealStatusEffect(target, xi.effectFlag.ERASABLE)
+        if erasableEffect ~= 0 then
+            transferCount = transferCount + 1
+        else
+            break  -- Break the loop if no more erasable effects are available
```

Comment:
I dont understand what this is doing or why. Could you explain it to me?

Reply:
After iterating through the "base" Sacrifice enfeebling effects that it can remove, the code iterates through the erasable effects. In Afflatus Solace stance, a White Mage can transfer up to 7 effects so this will continue the loop after going through the initial list to look for anything that can be erased.